### PR TITLE
SOF-6631: implementation of Hubbard card context provider

### PR DESCRIPTION
**Pull request by pranabdas**

### CODE REVIEW

File: src/context/providers/HubbardContextProvider.js [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+import { mix } from "mixwith";
+
+const defaultHubbardConfig = {
+    atomicSpecies: "",
+    atomicOrbital: "2p",
+    hubbardUValue: 0.01,
+};
+
+export class HubbardContextProvider extends mix(JSONSchemaFormDataProvider).with(
+    ApplicationContextMixin,
```
Comment (spectre007) [inline src/context/providers/HubbardContextProvider.js:17]: We're not using application information here, so `ApplicationContextMixin` can be removed. Same goes for the static `Application` variable below

File: src/context/providers/HubbardContextProvider.js [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+            description: "Hubbard parameters for DFT+U calculation.",
+            type: "array",
+            items: {
+                type: "object",
+                properties: {
+                    atomicSpecies: {
+                        type: "string",
+                        title: "Atomic species",
+                        enum: this.uniqueElements,
+                        default: this.uniqueElements[0],
```
Comment (spectre007) [inline src/context/providers/HubbardContextProvider.js:65]: let's only access the zeroth element if the list is not empty, perhaps something like this:
```javascript
default: this.uniqueElements?.length ? this.uniqueElements[0] : "",
```

File: src/context/providers/HubbardContextProvider.js [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+    ApplicationContextMixin,
+    MaterialContextMixin,
+) {
+    static Application = Application;
+
+    static Material = Made.Material;
+
+    constructor(config) {
+        super(config);
+        this.uniqueElements = this.material.Basis.uniqueElements;
```
Comment (spectre007) [inline src/context/providers/HubbardContextProvider.js:26]: to be safe, we should initialize as empty list if there is an issue with the material, for instance:
```javascript
this.uniqueElements = this.material?.Basis?.uniqueElements || [];
```

**Review by spectre007 — COMMENTED** [review]
minor changes suggested, otherwise looks good 👍

File: src/context/providers/HubbardContextProvider.js [hunk 4]
```diff
@@ -0,0 +1,74 @@
+import { JSONSchemaFormDataProvider, MaterialContextMixin } from "@exabyte-io/code.js/dist/context";
+import { Made } from "@exabyte-io/made.js";
+import { mix } from "mixwith";
+
+const defaultHubbardConfig = {
+    atomicSpecies: "",
+    atomicOrbital: "2p",
+    hubbardUValue: 0.01,
```
Comment (timurbazhirov) [inline src/context/providers/HubbardContextProvider.js:8]: maybe just `U`

**Review by timurbazhirov — APPROVED** [review]

**Review by timurbazhirov — APPROVED** [review]

**Pull request merged**